node/core/backing/src/lib.rs

```diff
@@ -0,0 +1,1454 @@
+// the code. So this does the necessary conversion.
+fn primitive_statement_to_table(s: &SignedFullStatement) -> TableSignedStatement {
+	let statement = match s.payload() {
+		Statement::Seconded(c) => TableStatement::Candidate(c.clone()),
+		Statement::Valid(h) => TableStatement::Valid(h.clone()),
+		Statement::Invalid(h) => TableStatement::Invalid(h.clone()),
+	};
+
+	TableSignedStatement {
+		statement,
+		signature: s.signature().clone(),
+		sender: s.validator_index(),
+	}
+}
+
+// finds the first key we are capable of signing with out of the given set of validators,
+// if any.
+fn signing_key(validators: &[ValidatorId], keystore: &KeyStorePtr) -> Option<ValidatorPair> {
+	let keystore = keystore.read();
+	validators.iter()
+		.find_map(|v| {
+			keystore.key_pair::<ValidatorPair>(&v).ok()
+		})
+		.map(|pair| pair)
+}
+
+impl CandidateBackingJob {
+	/// Run asynchronously.
+	async fn run(mut self) -> Result<(), Error> {
+		while let Some(msg) = self.rx_to.next().await {
+			match msg {
+				ToJob::CandidateBacking(msg) => {
+					self.process_msg(msg).await?;
+				}
+				_ => break,
+			}
+		}
+
+		Ok(())
+	}
+
+	async fn issue_candidate_invalid_message(
+		&mut self,
+		candidate: AbridgedCandidateReceipt,
+	) -> Result<(), Error> {
+		self.tx_from.send(FromJob::CandidateSelection(
+				CandidateSelectionMessage::Invalid(self.parent, candidate)
+		)).await?;
+
+		Ok(())
+	}
+
+	/// Validate the candidate that is requested to be `Second`ed and distribute validation result.
+	async fn validate_and_second(
+		&mut self,
+		candidate: AbridgedCandidateReceipt,
+		pov: PoVBlock,
+	) -> Result<ValidationResult, Error> {
+		let valid = self.request_candidate_validation(candidate.clone(), pov).await?;
+		let statement = match valid {
+			ValidationResult::Valid => {
+				// make PoV available for later distribution. Send data to the availability
+				// store to keep. Sign and dispatch `valid` statement to network if we
+				// have not seconded the given candidate.
+				Statement::Seconded(candidate)
+			}
+			ValidationResult::Invalid => {
+				let candidate_hash = candidate.hash();
+				self.issue_candidate_invalid_message(candidate).await?;
+				Statement::Invalid(candidate_hash)
+			}
+		};
+
+		if let Some(signed_statement) = self.sign_statement(statement) {
+			self.import_statement(signed_statement.clone()).await?;
+			self.distribute_signed_statement(signed_statement).await?;
+		}
+
+		Ok(valid)
+	}
+
+	fn get_backed(&self) -> Vec<NewBackedCandidate> {
+		let proposed = self.table.proposed_candidates(&self.table_context);
+		let mut res = Vec::with_capacity(proposed.len());
+
+		for p in proposed.into_iter() {
+			let TableAttestedCandidate { candidate, validity_votes, .. } = p;
+
+			let (ids, validity_votes): (Vec<_>, Vec<_>) = validity_votes
+						.into_iter()
+						.map(|(id, vote)| (id, vote.into()))
+						.unzip();
+
+			let mut validator_indices = BitVec::with_capacity(
+				self.table_context.validators.len(),
```

Comment:
Note that the `validator_indices` in `BackedCandidate are (quoting from implementer's guide)

```rust
  // the indices of validators who signed the candidate within the group. There is no need to include
  // bit for any validators who are not in the group, so this is more compact.
  // The number of bits is the number of validators in the group.
  //
  // the group should be apparent from context.
```

So if our validators are `[a, b, c, d]`, and the group members are `[3, 2]` (group assignments are shuffled and thus can be out-of-order), the bitfield will have length 2 where the first bit corresponds to validator index 3 (validator d) and the 2nd corresponds to validator index 2 (validator c).

This code seems to set the bits based on the index of the validator in the entire validator set.